lib/rex/post/meterpreter/packet_response_waiter.rb

```diff
@@ -57,25 +64,14 @@ def notify(response)
   # If the interval is -1 we can wait forever.
   #
   def wait(interval)
-    if( interval and interval == -1 )
-      while(not self.done)
-        ::IO.select(nil, nil, nil, 0.1)
-      end
-    else
-      begin
-        Timeout.timeout(interval) {
-          while(not self.done)
-            ::IO.select(nil, nil, nil, 0.1)
-          end
-        }
-      rescue Timeout::Error
-        self.response = nil
-      end
-    end
+    interval = nil if interval and interval == -1
+    self.mutex.synchronize {
```

Comment:
Good spot. Checking the the response should be done within the critical section though. Pushing a new commit for this now.